app/src/main/java/com/github/se/bootcamp/ui/screens/SettingsScreen.kt

```diff
@@ -0,0 +1,143 @@
+package com.github.se.bootcamp.ui.screens
+
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material.icons.outlined.ArrowBack
+import androidx.compose.material.icons.outlined.Edit
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.se.bootcamp.ui.navigation.NavigationActions
+
+@Composable
+fun SettingsScreen(profilePictureUrl: String?,
+                   navigationActions: NavigationActions) {
+    var username by remember { mutableStateOf("Test Name 1") }
+    var showEditOptions by remember { mutableStateOf(false) }
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            //.background(Color.DarkGray)
+            .padding(16.dp),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.spacedBy(64.dp)
+    ) {
+
+        // Back Button
+        Box(
+            modifier = Modifier
+                .fillMaxWidth() // Occupe toute la largeur disponible
+                .align(Alignment.Start) // Aligne uniquement l'icône à gauche
+        ) {
+            IconButton(onClick = { navigationActions.goBack() }) {
+                Icon(
+                    imageVector = Icons.Default.ArrowBack,
+                    contentDescription = "Back",
+                    tint = Color(0xFF05A9FB)
+                )
+            }
+        }
+        // Editable Username
+
+            Row(
+                modifier = Modifier
+                    .border(2.dp, Color.Gray, RoundedCornerShape(8.dp))
+                    .clip(RoundedCornerShape(8.dp))
+                    .background(Color(0xFFDCF1FA))
+                    .clickable { showEditOptions = true }
+                    .padding(horizontal = 16.dp, vertical = 8.dp),
+                verticalAlignment = Alignment.CenterVertically,
+                horizontalArrangement = Arrangement.Center // Centrer le contenu dans la Row
+
+                ) {
+                Icon(
+                    imageVector = Icons.Outlined.Edit,
+                    contentDescription = "Edit Username",
```

Comment:
It's a good habit to develop to avoid writing hard coded string in the code, and favor the use of String.xml resources. This will makes it easier to later [localize](https://developer.android.com/guide/topics/resources/localization) the application. It also makes the app more maintainable as it is easy to modify one text, without having to remember to modify every places that are displaying the hard coded text.